src/Appium.Net/Appium/Android/UiAutomator/AndroidUiScrollable.cs

```diff
@@ -41,14 +42,316 @@ public AndroidUiScrollable()
+            return new TerminatedStatementBuilder(forkedBuilder);
+        }
+
+        /// <summary>
+        /// Searches for a child element in the present scrollable container. The search first looks for a
+        /// child element that matches the selector you provided, then looks for the content-description in
+        /// its children elements. If both search conditions are fulfilled, the method returns a {@ link UiObject}
+        /// representing the element matching the selector (not the child element in its subhierarchy containing
+        /// the content-description). By default, this method performs a scroll search.
+        /// Maps to the UiScrollable.getChildByDescription(UiSelector, String, boolean) method.
+        /// </summary>
+        /// <param name="uiSelector">
+        /// UiSelector for a child in a scollable layout element
+        /// </param>
+        /// <param name="description">
+        /// Content-description to find in the children of the childPattern match (may be a partial match)
+        /// </param>
+        /// <param name="allowScrollSearch">
+        /// Set to true if scrolling is allowed
+        /// </param>
+        /// <remarks>https://developer.android.com/reference/android/support/test/uiautomator/UiScrollable#getchildbydescription_1</remarks>
+        public TerminatedStatementBuilder GetChildByDescription(AndroidUiSelector uiSelector, string description, bool allowScrollSearch = true)
+        {
+            if (uiSelector == null)
+                throw new ArgumentNullException(nameof(uiSelector));
+
+            if (description == null)
+                throw new ArgumentNullException(nameof(description));
+
+            var forkedBuilder = new StringBuilder(_builder.ToString());
+            forkedBuilder.AppendFormat(".getChildByDescription({0}, \"{1}\", {2})", 
+                uiSelector.Build(), 
+                description,
+                allowScrollSearch.ToString().ToLowerInvariant());
+            return new TerminatedStatementBuilder(forkedBuilder);
+        }
+
+        /// <summary>
+        /// Searches for a child element in the present scrollable container that matches the selector you provided.
+        /// The search is performed without scrolling and only on visible elements.
+        /// Maps to the UiScrollable.getChildByInstance(UiSelector, int) method.
+        /// </summary>
+        /// <param name="uiSelector"></param>
+        /// <param name="instance"></param>
+        /// <remarks>https://developer.android.com/reference/android/support/test/uiautomator/UiScrollable#getchildbyinstance</remarks>
+        public TerminatedStatementBuilder GetChildByInstance(AndroidUiSelector uiSelector, int instance)
+        {
+            if (uiSelector == null)
+                throw new ArgumentNullException(nameof(uiSelector));
+
+            if (instance < 0)
+                throw new ArgumentOutOfRangeException(nameof(instance), "Must be 0 (zero) or greater");
```

Comment:
this check could be extracted to a separate helper method, for example `requirePositiveInteger`

Reply:
Since the non-negative guards exist in both new classes, should I extract the guard check to a new internal class? Static?

Reply:
it's up to you, I'm not an expert here. The main purpose is to avoid unnecessary code duplication